Automattic/jetpack
# Add tracking for the contact form block

Right now we can track how many contact forms are submitted, but it's not possible to distinguish which ones are from the shortcode and which ones are from the block. This adds a tracks event specifically for the block, so that we can track it independently of the shortcode.

#### Changes proposed in this Pull Request:

* Relies on the `hasFormSettingsSet` attribute which is set by the block
* When this attribute exists we add a hidden field to the form with the name `is_block`
* If this hidden field is submitted we then add a value to the `extra_values` array
* If `is_block` is set in the `extra_values` array, then send a `jetpack_contact_form_gutenberg_block_message_sent` event to Tracks

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Adding features to Jetpack

#### Testing instructions:
* Create a new post and add a contact form block
* From the front end of your site, submit the contact form
* Go to Tracks and search for the event `jetpack_contact_form_gutenberg_block_message_sent`
* Now create another post and add the contact form shortcode
* From the front end of your site, submit the contact form
* Go to Tracks and search for the event `jetpack_contact_form_gutenberg_block_message_sent` - you should not see it.
